apis/v1beta1/vspherefailuredomain_types.go

```diff
@@ -81,12 +82,66 @@ type Topology struct {
 	// +optional
 	Networks []string `json:"networks,omitempty"`
 
+	// NetworkConfigurations is a list of network configurations within this failure domain.
+	// +optional
+	NetworkConfigurations []NetworkConfiguration `json:"networkConfigs,omitempty"`
```

Comment:
```suggestion
	//+listType=map
	//+listMapKey=name
	NetworkConfigurations []NetworkConfiguration `json:"networkConfigs,omitempty"`
```

This should also make the NetworkName key a unique identifier. Or is there a reason why it should be empty?